tephra-hbase-compat-0.96/src/main/java/co/cask/tephra/hbase96/coprocessor/TransactionProcessor.java

```diff
@@ -173,11 +175,15 @@ public void preDelete(ObserverContext<RegionCoprocessorEnvironment> e, Delete de
     Put deleteMarkers = new Put(delete.getRow(), delete.getTimeStamp());
     for (byte[] family : delete.getFamilyCellMap().keySet()) {
       List<Cell> familyCells = delete.getFamilyCellMap().get(family);
-      int cellSize = familyCells.size();
-      for (int i = 0; i < cellSize; i++) {
-        Cell cell = familyCells.get(i);
-        deleteMarkers.add(CellUtil.cloneFamily(cell), CellUtil.cloneQualifier(cell), cell.getTimestamp(),
-            HConstants.EMPTY_BYTE_ARRAY);
+      if (isFamilyDelete(familyCells)) {
+        deleteMarkers.add(family, TxConstants.FAMILY_DELETE_QUALIFIER, HConstants.EMPTY_BYTE_ARRAY);
```

Comment:
No, there is no reliable way to distinguish this at the moment.  In the future, we can use cell tags for this, but cell tags require HFile format v3, which is _not_ enabled by default in HBase 0.98.

Honestly, using our own delete markers should be viewed as a stop-gap measure.  I don't think we should invest much more effort in them beyond this current change.  In my opinion, we would be better served by getting the changes we need in HBase to support native HBase deletes.